component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/totp/util/TOTPUtilTest.java

```diff
@@ -601,4 +600,61 @@ public IObjectFactory getObjectFactory() {
 
         return new PowerMockObjectFactory();
     }
+
+    @Test(description = "Test case for getProcessedClaimValue()", dataProvider = "processedClaimValueTestDataProvider")
+    public void testGetProcessedClaimValue(String claimURI, Map<String, String> claimProperties, String claimValue,
+                                           String expectedClaimValue) throws Exception {
+
+//        1st approach
+//        spy(TOTPUtil.class);
+//        given(TOTPUtil.getClaimProperties("testDomain", claimURI)).willReturn(claimProperties);
+
+//        2nd approach
+//        doReturn(claimProperties).when(totpUtil).getClaimProperties("testDomain", claimURI);
+//        when(totpUtil.getClaimProperties("testDomain", claimURI)).thenReturn(claimProperties);
+
+//        3rd approach
+//        mockStatic(TOTPUtil.class);
+//        TOTPUtil totpUtilSpy = spy(new TOTPUtil());
+//        when(TOTPUtil.getClaimProperties("testDomain", "claimURI")).thenReturn(claimProperties);
+//        String processedClaimValue = totpUtilSpy.getProcessedClaimValue(claimURI, claimValue, "testDomain");
+
+//        4th approach
+//        TOTPUtil totpUtilMocked = Mockito.mock(TOTPUtil.class, Mockito.CALLS_REAL_METHODS);
+
+//        String processedClaimValue = totpUtil.getProcessedClaimValue(claimURI, claimValue, "testDomain");
+//
+//        // Verify the result
+//        assertEquals(expectedClaimValue, processedClaimValue);
+    }
+
+    @DataProvider(name = "processedClaimValueTestDataProvider")
+    public static Object[][] getProcessedClaimValueTestData() {
+
+        HashMap<String, String> claimProperties_1 = new HashMap<>();
+        claimProperties_1.put("EnableEncryption", "true");
+
+        HashMap<String, String> claimProperties_2 = new HashMap<>();
+        claimProperties_2.put("EnableEncryption", "false");
+
+        HashMap<String, String> claimProperties_3 = new HashMap<>();
+
+
+        return new Object[][]{
+                {TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL, claimProperties_1, "AER2BRI0LK4XCSC1", "AER2BRI0LK4XCSC1"},
+                {TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL, claimProperties_1, "", ""},
+                {TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL, claimProperties_2, "AER2BRI0LK4XCSC1", "AER2BRI0LK4XCSC1"},
+                {TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL, claimProperties_2, "", ""},
+                {TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL, claimProperties_3, "", ""},
+                {TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL, claimProperties_3, "AER2BRI0LK4XCSC1", "eyJjIjoiZXlKamFYQ" +
```

Comment:
What is this key ?

Reply:
This will no longer be needed as we are not adding the unit test with this PR.